pennylane/transforms/dynamic_one_shot.py

```diff
@@ -194,18 +205,26 @@ def init_auxiliary_tape(circuit: qml.tape.QuantumScript):
                 new_measurements.append(SampleMP(obs=m.obs))
             else:
                 new_measurements.append(m)
-    for op in circuit:
-        if is_mcm(op):
+    new_operations = []
+    for op in circuit.operations:
+        if isinstance(op, MidMeasureMP):
             new_measurements.append(qml.sample(MeasurementValue([op], lambda res: res)))
-
+        if "MidCircuitMeasure" in str(type(op)):
```

Comment:
> This needs to be if because the block changes both new_measurements and new_operations.

Oh my bad, but it's still confusing because the op can't be both `MidMeasureOp` and `MidCircuitMeasure` right? At least these two instructions should be exclusive (which is currently not enforced by the code structure, in fact I think with Mudit's inheritance PR both of these branches would be triggered):

```py
new_measurements.append(qml.sample(MeasurementValue([op], lambda res: res)))
new_measurements.append(qml.sample(op.out_classical_tracers[0]))
```

So while I'm not sure exactly how these three branches are supposed to be executing, I still think it would be better that the two statements I listed be guaranteed to be exclusive.